backend/swift/swift.go

```diff
@@ -1148,9 +1149,22 @@ func (o *Object) updateChunks(in0 io.Reader, headers swift.Headers, size int64,
 		err = o.fs.pacer.CallNoRetry(func() (bool, error) {
 			var rxHeaders swift.Headers
 			rxHeaders, err = o.fs.c.ObjectPut(o.fs.segmentsContainer, segmentPath, segmentReader, true, "", "", headers)
+			if err == nil {
+				segmentInfos = append(segmentInfos, segmentPath)
+			}
 			return shouldRetryHeaders(rxHeaders, err)
 		})
 		if err != nil {
+			if len(segmentInfos) > 0 {
```

Comment:
This bit of code is duplicated - can you factor it out into a function (it could be a local function) and call it in both places?

Reply:
Hi @ncw ,
I’ve already been to refactor the block code what you mention above to func deleteChunks